src/main/java/org/traccar/protocol/PretraceProtocolDecoder.java

```diff
@@ -90,7 +92,16 @@ protected Object decode(
         position.set(Position.KEY_HDOP, parser.nextInt(0));
         position.set(Position.KEY_RSSI, parser.nextInt(0));
 
-        parser.next(); // state
+        String stateValue = parser.next(); // state
```

Comment:
We usually save state as a single value. Also, if you want to break it down, you should probably use standard prefixes. There are also a bunch of formatting issues here.

Reply:
Store it in whatever value you want. Not storing it is not a good option I think. This is a bitfield with different states - I did not fiend a good existing standard prefix. If you know a good standard prefix please use it.